dockers/docker-database/Dockerfile.j2

```diff
@@ -32,9 +32,21 @@ RUN sed -ri 's/^(save .*$)/# \1/g;                       \
              s/^logfile .*$/logfile ""/;                 \
              s/^# syslog-enabled no$/syslog-enabled no/; \
              s/^# unixsocket/unixsocket/;                \
+             s%pidfile .*%pidfile /var/run/redis/redis.pid%; \
              s/^client-output-buffer-limit pubsub [0-9]+mb [0-9]+mb [0-9]+/client-output-buffer-limit pubsub 0 0 0/ \
             ' /etc/redis/redis.conf
 
+RUN sed -ri 's%DAEMON_ARGS=.*%DAEMON_ARGS=/etc/`basename ${0}`/`basename ${0}`.conf%;  \
+             s/NAME=.*/NAME=`basename ${0}`/;                                          \
```

Comment:
>basename ${0} [](start = 29, length = 13)

Could you give an example for `basename ${0}`?

Reply:
redis-server is a script, when we using basename ${0} inside, actually it takes the name of the executed script.
for each instance, we keep one script(copied from redis-server) named with the instance name.
Then when we run below command, basename is translated to redis , redis_inst_002, redis_inst_003, then we can get the correct, pid, conf files


root@PSW-7005:/# /etc/init.d/redis  status
[ ok ] redis is running.
root@PSW-7005:/# /etc/init.d/redis_ins_002  status
[ ok ] redis_ins_002 is running.
root@PSW-7005:/# /etc/init.d/redis_ins_003  status
[ ok ] redis_ins_003 is running.
root@PSW-7005:/#